src/gt4py/_core/locking.py

```diff
@@ -0,0 +1,26 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import pathlib
+from typing import ContextManager, Protocol
+
+import filelock
+
+
+class _LockConstructor(Protocol):
+    def __call__(self, path: str, /) -> ContextManager: ...
+
+
+_default_lock: _LockConstructor = filelock.FileLock
```

Comment:
Maybe document that this is private/experimental API for now but might be made public in the future.